tests/commands/method.py

```diff
@@ -19,10 +19,164 @@
+        # Test the default method.
+        handler = self.handler()
+        self._test_seen(handler, None)
+
+        # Try to access correct attribute.
+        handler = self.handler()
+        handler.method.method = "nfs"
+
+        setattr(handler.method, "server", "1.2.3.4")
+        handler.method.server = "1.2.3.4"
+        getattr(handler.method, "server")
+        if handler.method.server:
+            pass
+
+        # Try to access wrong attribute.
+        handler = self.handler()
+        handler.method.method = "url"
+
+        with self.assertRaises(AttributeError):
+            setattr(handler.method, "server", "1.2.3.4")
+
+        with self.assertRaises(AttributeError):
+            handler.method.server = "1.2.3.4"
+
+        with self.assertRaises(AttributeError):
+            getattr(handler.method, "server")
+
+        with self.assertRaises(AttributeError):
+            if handler.method.server:
+                pass
+
+        # Try to set a nonexistent method.
+        handler = self.handler()
+        with self.assertRaises(AttributeError):
+            handler.method.method = "???"
+
+        # Try to get and set a nonexistent attribute.
+        handler = self.handler()
+        handler.method.method = "url"
+
+        with self.assertRaises(AttributeError):
+            getattr(handler.method, "nonexistent_attribute")
+
+        with self.assertRaises(AttributeError):
+            if handler.method.nonexistent_attribute:
+                pass
+
+        with self.assertRaises(AttributeError):
+            setattr(handler.method, "nonexistent_attribute", "some_value")
+
+        with self.assertRaises(AttributeError):
+            handler.method.nonexistent_attribute = "some_value"
+
+        # Try to call a nonexistent method.
+        handler = self.handler()
+        with self.assertRaises(AttributeError):
+            handler.method.nonexistent_method()
+
+        # Test hasattr.
+        handler = self.handler()
+        self.assertEqual(hasattr(handler.method, "seen"), True)
+        self.assertEqual(hasattr(handler.method, "server"), False)
+        self.assertEqual(hasattr(handler.method, "nonexistent"), False)
+
+        handler.method.method = "nfs"
+        self.assertEqual(hasattr(handler.method, "seen"), True)
+        self.assertEqual(hasattr(handler.method, "server"), True)
+        self.assertEqual(hasattr(handler.method, "nonexistent"), False)
+
+        # Test copy and deepcopy.
+        handler = self.handler()
+        handler.method.method = "url"
+        handler.method.url = "http://domain.com"
+
+        method2 = copy.copy(handler.method)
+        self.assertEqual(method2.method, "url")
+        self.assertEqual(method2.url, "http://domain.com")
+
+        # FIXME: Fails with "TypeError: cannot deepcopy this pattern object"
+        #method3 = copy.deepcopy(handler.method)
+        #self.assertEqual(method3.method, "url")
+        #self.assertEqual(method3.url, "http://domain.com")
```

Comment:
Please remove these lines and write here a `TODO` instead. It will be less disrupting for reading.